### ISSUE TITLE
timezone issues still there

### ISSUE THREAD

**Issue by pulketo** [loadavg/loadavg#23]
The graphics for your timezone are not OK, for example
GMT-6 at 16:30 we see our graphics on 22:30...
![loadavg](https://cloud.githubusercontent.com/assets/337881/6051525/7a2c7d92-ac8e-11e4-983b-22cc1c2ee3ba.png)

**Comment by loadavg** [comment]
Thanks for bringing this up - will look into it and see what's happening.

Can you let me know if your server in the same timeZone as the pc you are using to access loadavg ?

**Comment by pulketo** [comment]
Yes, i'm testing on localhost so TZ settings are the same, temporary solution while i figure out how to set this globally is:
Adding detection of timezone (via browser) on each module's javascript (example: lib/modules/Disk/disk.js) line 56
xaxis: {mode: 'time', minTickSize: ................................................
with this:
xaxis: {mode: 'time', timezone: "browser", minTickSize: ............................................
seems to fix time for disk usage

**Comment by hamilton5** [comment]
I posted here:
https://github.com/loadavg/loadavg/issues/2

I still have this problem.

**Comment by loadavg** [comment]
Looking into this now - since we are setting timezone via browser the fix here makes sense. will try it out and see...

**Comment by loadavg** [comment]
so the browser fix is just a patch really, we need to pass the timezone to the chart as they do in the source here ...

http://www.flotcharts.org/flot/examples/axes-time-zones/

going to look into it...

**Comment by loadavg** [comment]
so that wont work either really - did some digging and the issue is that there are really 3 timezones:

1. your servers timezone
2. the timezone set in php (should match 1 but doesn't always)

3. your browsers timezone

The logger logs 24 hours of data based on the timezone set in PHP from 0 to 24 hours - if this is not set then PHP defaults to UTC (GMT). If you set this in loadavg it uses the timezone value you set it to as a override but this will affeact log data  as well so shouldnt be changed really.

Then we have the browsers timezone to deal with which can be used to reformat the periods log data to the timezone you are in based on the xaxis to make it easy to understand.

So we are going to break this out into 2 settings - one for logging and one for viewing. 

The timezone for logging with be managed under the logger - it will use the PHP/system value unles you override it - and really you shouldnt right ? your server should be set up correctly.

The timezone for viewing will allow you to reformat the x-axis time data in your chart - without moving the chart data around - so you can view it based on your local time if you like.

Let me know if this sounds like the way it should work.

**Comment by loadavg** [comment]
Fixed this - if you pull 2.2 it should work better now though needs some cleaning up...

# git checkout -b 2.2
# git pull origin 2.2

If you are updating from 2.2 git then you will need to make changes to settings.ini to add new options

see the new default settings here
https://github.com/loadavg/loadavg/blob/2.2/app/config/default_settings.ini.php

added under [settings]

clienttimezone = "America/New_York"
timezonemode = "UTC"

Timezone settings for the logger have now been moved to the logger settings page - should default to how PHP is set up at install time but can be overriden there if need be.

We added a option under main settings to let you choose display timezone - UTC or browser (or optionally to override with the timezone you want - but this isnt finished yet) 

All data and view spans are now in UTC based on the logger daily span and the X axis is just reformatted based on the viewing timezone (browser etc) to reflect the correct times...

**Comment by hamilton5** [comment]
I don't understand why it's so difficult to match up the axis label.. http://imgur.com/BZw8RTD

The browser timezone seems like a convenience, If one wanted to display in their own timezone. Why would someone have a different PHP timezone?

I tried two timezonemodes UTC and EST, clienttimezone america/new_york is ok, but I did not see the x axis change.

**Comment by pulketo** [comment]
Well most
El feb 11, 2015 6:55 PM, "HaMiLToN5" <notifications@github.com> escribió:
>
> I don't understand why it's so difficult to match up the axis label..
http://imgur.com/BZw8RTD
>
> The browser timezone seems like a convenience, If one wanted to display
in their own timezone. Why would someone have a different PHP timezone?
>
> I tried two timezonemodes UTC and EST, clienttimezone america/new_york is
ok, but I did not see the x axis change.

**Comment by loadavg** [comment]
Sorry - checked in the rest of the fix now and it should all work smoothly- some modules werent updated with the new code so didnt show anything but all should be ok now.

Yes, timezones are tricky as really there affect us in 3 places as mentioned - server, php (where the script is run) and then your local browser! Gets messy.

You really just need to:

a. Set your logger's timezone in page 'settingslogger' to the timezone your server is in 

This is done at install time automatically and should line up with your servers time/clock. You can override this if you want but should never change it once set!

b. Set your browser timezone in page 'settings' to 'browser' (will be the default)

And you should see the log data remapped to your browsers timezone for you.

You can set the browser timezone to UTC for a 1:1 correlation to the server (so you can see the raw logs) or if you set to 'override' you can then manually set the browsers timezone there as well (will hide this when not selected as override) - not necessary really but there if its needed for some reason (like different people looking at the logs at the same time from different locations)

**Comment by hamilton5** [comment]
spot on, working well now, thank you.

**Comment by pulketo** [comment]
still something is missing...
check my screenshot 

thanks :)
![loadavg-2015-02-26_16-08-05](https://cloud.githubusercontent.com/assets/337881/6403150/f9efe4ba-bdd2-11e4-8863-10c0b3ea36cc.jpg)

**Comment by loadavg** [comment]
sorry - grab 2.2 and then you need to set your Timezone under System Settings to 'browser' - you are currently set to UTC. 

my bad on the high/low values need to take a look there and see whats up...

**Comment by laitosko** [comment]
hi.

using 2.2 and having some problems with timezone as well. using override and europe tz but everything's still in 12hour am/pm format. system is set correctly. any ideas?
